tests/test_runlib.py

```diff
@@ -642,6 +707,34 @@ def test_create_metadata_with_expected_product(self):
     self.assertEqual(list(link.signed.products.keys()), [self.test_product])
     os.remove(self.link_name)
 
+  def test_create_metadata_with_expected_product_in_target_directory(self):
+    """Test record stop records expected product in target directory """
+    tmp_dir = os.path.realpath(tempfile.mkdtemp(dir=os.getcwd()))
+    in_toto_record_start(self.step_name, [], self.key)
+    in_toto_record_stop(self.step_name, [self.test_product], self.key,
+        metadata_directory=tmp_dir)
+    link_path = os.path.join(tmp_dir, self.link_name)
+    link = Metablock.load(link_path)
+    self.assertEqual(list(link.signed.products.keys()), [self.test_product])
```

Comment:
I'm still not sure I understand this test in the context of `metadata_directory`. Could you go over the reasoning, @SolidifiedRay? :slightly_smiling_face:

Reply:
I'd have expected it to be similar to `test_in_toto_run_with_metadata_directory`.

Reply:
For `test_in_toto_run_with_metadata_directory`, `in_toto_run` itself will return a link, so I can compare that link with the `link_dump`. But I think `in_toto_record_stop` will not return a link, so I am not sure what to compare. 
For example, I can do `link = in_toto_run()` and compare this `link` with `link_dump`.
However, I am not able to do something like `link = in_toto_record_stop()`
Do you have any suggestions for this? @adityasaky

Reply:
Ah I see. I think the test which runs `in-toto-record` twice, with and without `metadata_directory`, suffices. Maybe we can dump this one.